### PR TITLE
[CR] Ability to open safes (etc.) by drilling and using acetylene torch

### PR DESCRIPTION
**Pull request by nexusmrsep**
#### Summary
```SUMMARY: Features "Open safes (etc.) by drilling and using acetylene torch"```

#### Purpose of change
Answer to @DeltaEpsilon7787  idea for resolving #27667

#### Describe the solution

- Level 3 drills (power drills) can be used to drill door locks and safes (including jammed, brute force)
- Acetylene torch can be used to crack safes (including jammed, brute force)

- torch use can be dangerous to contents
- some expertise in mechanics is required
- new safe types: gunsafe open and gunsafe closed (but not locked) for visual purposes in tilesets (pinging @SomeDeadGuy )
- added hints to examining safes to tell players that there may be more then one obvious way to get inside
- migrated `f_gun_safe_ml` `furn_id` to `f_gunsafe_ml` to unify naming convention plus added some migration code for the savegames for that reason

#### Describe alternatives you've considered
I made a huge menu for examining safes, then I got rid of it, for QoL reasons and sanity, since it would be too bloated with cross synchronization. Main reason for insane number of commits in this PR. Added hints instead. 

#### Additional context
TODO:
- [ ] find out why small chunk of migration code for furniture suddenly makes terrain saves partially unreadable for json reader
- [ ] do another round or two of tests
- [ ] listen to opinions and maybe do some tweaks where needed

### CODE REVIEW

**Comment by Wokko1** [comment]
I don't think it's a good idea to add separate tiles just for drill-opened safes (unless I'm misunderstanding). And drilling probably should take a long time, but it's a good idea overall to turn the game into Payday.

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -2551,6 +2577,76 @@ void activity_handlers::cracking_do_turn( player_activity *act, player *p )
...
+    const tripoint &pos = act->placement;
+    const ter_id type = g->m.ter( act->placement );
+    const furn_id furn = g->m.furn( act->placement );
+    ter_id new_type = t_concrete;
+    std::string open_message;
+    bool is_safe = false;
+
+    if( type == t_chaingate_l ) {
+        new_type = t_chaingate_c;
+    } else if( type == t_door_locked || type == t_door_locked_alarm ||
```
Comment (Night-Pryanik) [inline src/activity_handlers.cpp]: Maybe make it so alarm sets off whilst drilling?

File: src/iuse.cpp [hunk 2]
```diff
@@ -3042,6 +3042,65 @@ int iuse::jackhammer( player *p, item *it, bool, const tripoint & )
...
+        duration = 15_minutes;
+    } else if( furn == f_safe_l || furn == f_gunsafe_ml || furn == f_gunsafe_el ||
+               furn == f_gunsafe_mj ) {
+        if( p->get_skill_level( skill_mechanics ) < 2 ) {
+            p->add_msg_if_player( m_bad,
+                                  _( "Your expertise in mechanics is insufficient to properly drill the locking mechanism." ) );
+            return 0;
+        }
+        duration = std::max( 150_minutes - 20_minutes * ( p->get_skill_level( skill_mechanics ) - 3 ) -
+                             10_minutes * ( p->get_per() - 8 ), 30_minutes );
```
Comment (Night-Pryanik) [inline src/iuse.cpp]: Low perception (especially real low, such as 1 or 2) could make the process real long, up to three and a half hours. Just an observation.

**Comment by nexusmrsep** [comment]
@Wokko1 the reason behind it is that open/closed safe was also recently added, and gun safes (at least in DeadPeople tileset) have different textures (large boxes looking like vending machines) then normal safes (smaller box-like cubes). When locked gunsafe was converted to small open safe (via cracking etc.), the visual effect was terrible, so I decided that those safes deserve a separate look. It's not only for drilling, but I'll double check if other methods do the proper conversion too. 

Also - drilling takes a long time here, for safes especially, less for locks in doors.

File: src/iuse.cpp (hunk 2 above)
Comment (nexusmrsep) [reply src/iuse.cpp]: I've used existing math used in stethoscope action, and it seemed proper. Drilling thick metal is not a quick job when you're blind as a mole (low per).

**Comment by Hatfish** [comment]
Now can you make friendly NPCs say "_Guys, the thermal drill, go get it!_" whenever they spot a locked safe?

**Comment by Mcrone617** [comment]
So when you drill a door, it just unlocks it and then keeps the door intact right? So with this PR it will finally be possible to break into locked metal doors like the police station and keep the door around right?

**Comment by nexusmrsep** [comment]
@Mcrone617  exactly that.

**Comment by dissociativity** [comment]
Since the PR does't specify:
It does make noise right?

**Comment by kevingranade** [comment]
This is looking pretty stale. Are you planning on picking it up again soon?

**Comment by nexusmrsep** [comment]
Yes, I'm picking it up in a couple of days. Got involved in other PR and so it happened that it was left behind for quite some time. But I'm getting back to it soon, so don't close it, please.

**Comment by nexusmrsep** [comment]
Actually this fell so behind that resolving conflicts would be less optimal then doing a second approach in future, so I'm closing it now. Its free to pick up. Or if nobody picks it up, I may come back to it, but simply can't do it now.

**Pull request closed without merging**